wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -801,17 +753,58 @@ public function delete() {
 			}
 		}
 		
-		// 1st: Change user to guest, otherwise other the entire session, including
-		// all virtual sessions of the user will be deleted
-		$this->changeUser(new User(null));
+		// Delete session.
+		$sql = "DELETE FROM wcf".WCF_N."_".($this->isACP ? 'acp' : 'user')."_session
+			WHERE	sessionID = ?";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([$this->sessionID]);
 		
-		// 2nd: Actually remove session
-		/** @var \wcf\data\DatabaseObjectEditor $sessionEditor */
-		$sessionEditor = new $this->sessionEditorClassName($this->session);
-		$sessionEditor->delete();
+		// Delete legacy session.
+		if (!$this->isACP) {
+			$sql = "DELETE FROM wcf".WCF_N."_session
+				WHERE	sessionID = ?";
+			$statement = WCF::getDB()->prepareStatement($sql);
+			$statement->execute([$this->sessionID]);
+		}
+	}
+	
+	/**
+	 * Prunes expired sessions.
+	 */
+	public function prune() {
+		// Prevent the sessions from expiring while the development mode is active.
+		if (!ENABLE_DEBUG_MODE || !ENABLE_DEVELOPER_TOOLS) {
+			$sql = "DELETE FROM	wcf".WCF_N."_user_session
+				WHERE		lastActivityTime < ?";
+			$statement = WCF::getDB()->prepareStatement($sql);
+			$statement->execute([
+				TIME_NOW - self::ACP_SESSION_LIFETIME,
+			]);
+		}
 		
-		// disable update
-		$this->disableUpdate();
+		$sql = "DELETE FROM	wcf".WCF_N."_user_session
+			WHERE		lastActivityTime < ?
+				AND	userID IS NULL";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([
+			TIME_NOW - self::GUEST_SESSION_LIFETIME,
+		]);
+		
+		$sql = "DELETE FROM	wcf".WCF_N."_user_session
+			WHERE		lastActivityTime < ?
+				AND	userID IS NOT NULL";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([
+			TIME_NOW - self::USER_SESSION_LIFETIME,
+		]);
```

Comment:
Please combine these two queries to improve the performance. In situations with extreme amounts of session this query has caused significant bottlenecks through locks.